PyPortal_AlarmClock/code.py

```diff
@@ -12,18 +12,19 @@
 All text above must be included in any redistribution.
 """
 
-#pylint:disable=redefined-outer-name,no-member,global-statement
+#pylint:disable=redefined-outer-name,no-member,global-statement,no-self-use
+#pyline:disable=too-many-branches,too-many-statements,useless-super-delegation
 
 import time
 import json
 import board
+from secrets import secrets
 from adafruit_pyportal import PyPortal
 from adafruit_bitmap_font import bitmap_font
 from adafruit_display_text.text_area import TextArea
 from digitalio import DigitalInOut, Direction, Pull
 import analogio
 import displayio
-from secrets import secrets
 
 # Set up where we'll be fetching data from
 DATA_SOURCE = 'http://api.openweathermap.org/data/2.5/weather?q='+secrets['weather_location']
```

Comment:
Have you considered using the City ID here rather than name? If there are duplicate city names in a country it is hard to get the right one. id or zip gives more control.
for me - I live in Durham , NH , USA " but City name "Durham,US" does not work.
using "id" instead of "q"  with my city ID "508618"  gets me the right report

Reply:
Done